chia/wallet/wallet.py

```diff
@@ -363,13 +362,24 @@ async def _generate_unsigned_transaction(
             )
         elif exclude_coins is not None:
             raise ValueError("Can't exclude coins when also specifically including coins")
+
         assert len(coins) > 0
         self.log.info(f"coins is not None {coins}")
         spend_value = sum([coin.amount for coin in coins])
-
+        self.log.info(f"spend_value is {spend_value} and total_amount is {total_amount}")
         change = spend_value - total_amount
         if negative_change_allowed:
             change = max(0, change)
+        if change < 0 < fee:
+            fee_coins = await self.select_coins(
+                uint64(fee),
+                excluded_coin_amounts=exclude_coin_amounts,
+                exclude=([] if exclude_coins is None else list(exclude_coins)) + list(coins or []),
+            )
+            coins = coins.union(fee_coins)
+            spend_value = sum([coin.amount for coin in coins])
+            self.log.info(f"Updated spend_value is {spend_value} and total_amount is {total_amount}")
+            change = spend_value - total_amount
```

Comment:
Can you comment what is going on here, I'm not sure why we would do this (I don't think we have many code paths that hit negative change)

Reply:
right, wasn't sure where to put this, but when we generate offer coins, we don't include `fee` into it, then we call generated_signed tx and we pass `coins` to it so additional coins are not selected if we don't have enough coin. When fee is low, this is not a problem because we have enough change to use it for fee, but when fee is larger or user has a lot of small amount coins this errors out.